File: Platforms/SQ_Soil_Temperature/SQ_Soil_Temperature/src/cs/SQ_Soil_Temperature/SoilTemperatureExogenous.cs
using System;
using System.Collections.Generic;

public class SoilTemperatureExogenous 
{
    private double _meanTAir;
    private double _minTAir;
    private double _maxTAir;
    private double _dayLength;
    
        public SoilTemperatureExogenous() { }
    
    
    public SoilTemperatureExogenous(SoilTemperatureExogenous toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    _meanTAir = toCopy._meanTAir;
    _minTAir = toCopy._minTAir;
    _maxTAir = toCopy._maxTAir;
    _dayLength = toCopy._dayLength;
    }
    }
    public double meanTAir
        {
            get { return this._meanTAir; }
            set { this._meanTAir= value; } 
        }
    public double minTAir
        {
            get { return this._minTAir; }
            set { this._minTAir= value; } 
        }
    public double maxTAir
        {
            get { return this._maxTAir; }
            set { this._maxTAir= value; } 
        }
    public double dayLength
        {
            get { return this._dayLength; }
            set { this._dayLength= value; } 
        }
}

File: Platforms/SQ_Soil_Temperature/SQ_Soil_Temperature/src/java/SQ_Soil_Temperature/SoilTemperatureExogenous.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class SoilTemperatureExogenous
{
    private Double meanTAir;
    private Double minTAir;
    private Double maxTAir;
    private Double dayLength;
    
    public SoilTemperatureExogenous() { }
    
    public SoilTemperatureExogenous(SoilTemperatureExogenous toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.meanTAir = toCopy.getmeanTAir();
            this.minTAir = toCopy.getminTAir();
            this.maxTAir = toCopy.getmaxTAir();
            this.dayLength = toCopy.getdayLength();
        }
    }
    public Double getmeanTAir()
    { return meanTAir; }

    public void setmeanTAir(Double _meanTAir)
    { this.meanTAir= _meanTAir; } 
    
    public Double getminTAir()
    { return minTAir; }

    public void setminTAir(Double _minTAir)
    { this.minTAir= _minTAir; } 
    
    public Double getmaxTAir()
    { return maxTAir; }

    public void setmaxTAir(Double _maxTAir)
    { this.maxTAir= _maxTAir; } 
    
    public Double getdayLength()
    { return dayLength; }

    public void setdayLength(Double _dayLength)
    { this.dayLength= _dayLength; } 
    
}